bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/HorizonsDesignation.scala

```diff
@@ -0,0 +1,25 @@
+package edu.gemini.spModel.core
+
+import scalaz._, Scalaz._
+
+/** 
+ * Unique Horizons designation, which must be formatted differently depending on the object type.
+ * The `queryString` should yield a unique result.
+ */
+sealed abstract class HorizonsDesignation(val designation: String, val queryString: String) 
+  extends Product with Serializable
+
+object HorizonsDesignation {
+
+  final case class Comet(des: String)     extends HorizonsDesignation(des, s"DES=$des;CAP")
```

Comment:
This is the literal search string you send to Horizons. Andy and I talked to the JPL guy and figured it out after a lot of experimentation.
- For comets you need `DES=...;CAP` which says use the provided designation, with the current apparition (i.e., the newest ephemeris). So the current orbit of Halley's comet is `DES=1P;CAP`.
- For asteroids you must have `DES=` but cannot have `;CAP`. So 599 Luisa is `DES=1906 UJ`.
- And for major bodies like planets and their moons you cannot have either. So Mercury is just `1`.